00x_Exploring_The_National_Water_Model.ipynb

```diff
@@ -0,0 +1,382 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask_gateway import Gateway\n",
+    "# instantiate dask gateway\n",
+    "gateway = Gateway()\n",
+    "\n",
+    "# setup a cluster\n",
+    "options = gateway.cluster_options()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "options.conda_environment='users/pangeo'\n",
+    "options.profile = 'Medium Worker'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cluster = gateway.new_cluster(options)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cluster.adapt(minimum=4, maximum=30)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get the client for the cluster\n",
+    "client = cluster.get_client()\n",
+    "# view the client UI\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Open Zarr datasets in Xarray using a mapper from fsspec.  We use `anon=True` for free-access public buckets like the AWS Open Data Program, and `requester_pays=True` for requester-pays public buckets. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 's3://noaa-nwm-retro-v2-zarr-pds'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fs = fsspec.filesystem('s3', anon=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "ds = xr.open_dataset(fs.get_mapper(url), engine='zarr', chunks={},\n",
+    "                     backend_kwargs=dict(consolidated=True))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds.data_vars"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var='streamflow'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds[var]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f'Variable size: {ds[var].nbytes/1e12:.1f} TB')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Find the site with the largest streamflow on June 1, 2017"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "imax = ds[var].sel(time='2017-06-01 00:00:00').argmax().values"
```

Comment:
this was the last line I was able to compute on my little laptop. Aside from not having enough memory locally, I didn't have any issues running this on the `ogc` env.